lib/spack/spack/environment.py

```diff
@@ -707,6 +708,18 @@ def _read_manifest(self, f, raw_yaml=None):
         configuration = config_dict(self.yaml)
         self.concretization = configuration.get('concretization')
 
+        # Retrieve dev-build packages:
+        self.dev_specs = configuration['develop']
+        for name, entry in self.dev_specs.items():
+            # spec must include a concrete version
+            assert Spec(entry['spec']).version.concrete
+            # very basic regex match for a valid path, do not test existence
+            assert re.match(r'{0}?(.*{0})*.*'.format(os.sep),
```

Comment:
Will this regex match any string? I think `.` matches any character, so you may want to replace it with something like `[a-zA-Z0-9-_]` (alpha-numeric, the `-` and `_` characters) or Python may supply a character class for you.